adapters/repos/db/aggregator/filtered.go

```diff
@@ -35,49 +37,105 @@ func newFilteredAggregator(agg *Aggregator) *filteredAggregator {
 }
 
 func (fa *filteredAggregator) Do(ctx context.Context) (*aggregation.Result, error) {
-	out := aggregation.Result{}
+	if fa.params.Hybrid != nil {
+		return fa.hybrid(ctx)
+	}
 
-	// without grouping there is always exactly one group
-	out.Groups = make([]aggregation.Group, 1)
+	return fa.filtered(ctx)
+}
 
-	var (
-		allowList helpers.AllowList
-		foundIDs  []uint64
-		err       error
-	)
+func (fa *filteredAggregator) hybrid(ctx context.Context) (*aggregation.Result, error) {
+	sparseSearch := func() ([]*storobj.Object, []float32, error) {
+		kw, err := fa.buildHybridKeywordRanking()
+		if err != nil {
+			return nil, nil, fmt.Errorf("build hybrid keyword ranking: %w", err)
+		}
+
+		if fa.params.ObjectLimit == nil {
+			limit := hybrid.DefaultLimit
+			fa.params.ObjectLimit = &limit
+		}
 
-	if fa.params.Filters != nil {
-		s := fa.getSchema.GetSchemaSkipAuth()
-		allowList, err = inverted.NewSearcher(fa.store, s, fa.invertedRowCache, nil,
-			fa.Aggregator.classSearcher, fa.deletedDocIDs, fa.stopwords, fa.shardVersion).
-			DocIDs(ctx, fa.params.Filters, additional.Properties{},
-				fa.params.ClassName)
+		sparse, dists, err := fa.bm25Objects(ctx, kw)
 		if err != nil {
-			return nil, errors.Wrap(err, "retrieve doc IDs from searcher")
+			return nil, nil, fmt.Errorf("aggregate sparse search: %w", err)
 		}
+
+		return sparse, dists, nil
+	}
+
+	denseSearch := func(vec []float32) ([]*storobj.Object, []float32, error) {
+		if vec == nil {
+			return nil, nil, fmt.Errorf("must provide hybrid search vector if alpha > 0")
```

Comment:
I'm not too sure about exactly what Aggregate{} does, but in Get{}, the vector is optional and if not present, is calculated from the query parameter.  Is it possible to do the same thing here?

Reply:
This is actually still the case, although the logic exists outside the local usecase closure, and lives instead in the generic hybrid search logic ([see here](https://github.com/semi-technologies/weaviate/blob/feature/agg-hybrid-support/usecases/traverser/hybrid/searcher.go#L266-L284))

But currently, aggregation does not support any module providers, so this feature is unused by Aggregation until we decide to add support for modules in Aggregation queries